src/main/java/frc/robot/hardware/request/cansparkmax/SparkMaxRequest.java

```diff
@@ -0,0 +1,60 @@
+package frc.robot.hardware.request.cansparkmax;
+
+import com.revrobotics.CANSparkBase;
+import frc.robot.hardware.request.IRequest;
+
+import java.util.function.Function;
+
+public class SparkMaxRequest<T> implements IRequest<T> {
+
+	private final CANSparkBase.ControlType controlType;
+	private final int pidSlot;
+	private final Function<T, Double> feedforwardCalculator;
+	private final Function<T, Double> setPointToDoubleConverter;
+	private T setPoint;
+
+	SparkMaxRequest(
+		T setPoint,
+		CANSparkBase.ControlType controlType,
+		int pidSlot,
+		Function<T, Double> feedforwardCalculator,
+		Function<T, Double> setPointToDoubleConverter
+	) {
+		this.setPoint = setPoint;
+		this.controlType = controlType;
+		this.pidSlot = pidSlot;
+		this.feedforwardCalculator = feedforwardCalculator;
+		this.setPointToDoubleConverter = setPointToDoubleConverter;
+	}
+
+	SparkMaxRequest(T setPoint, CANSparkBase.ControlType controlType, int pidSlot, Function<T, Double> setPointToDoubleConverter) {
+		this(setPoint, controlType, pidSlot, CANSparkMAX -> 0.0, setPointToDoubleConverter);
+	}
+
+	@Override
+	public SparkMaxRequest<T> withSetPoint(T setPoint) {
+		this.setPoint = setPoint;
+		return this;
+	}
```

Comment:
Given a function name like `withSetPoint` I wouldn't expect `this` to be modified, but a new request object to be created. I do however see how the designed use case makes it so this makes no difference. I'll leave this at your discretion.